tests/test_engine.py

```diff
@@ -296,7 +303,16 @@ def test_close_engine_spiders_downloader(self):
         yield e.close()
         self.assertFalse(e.running)
         self.assertEqual(len(e.open_spiders), 0)
-
+    
+    @defer.inlineCallbacks
+    def test_crawler_with_error_in_start_requests(self):
+         self.run = CrawlerRun(ErrorInStartRequestsSpider)
+         yield self.run.run()
+         print(type(self.run.crawler.stats.get_stats()['finish_reason']))
+         # Check both as the latter one is produced for Python < 3.7
+         # https://bugs.python.org/issue30399 
```

Comment:
You could check the run-time Python version and use one check or the other depending on that.